### DISCUSSION TITLE
Disabling Limits in CRUD Columns

### DISCUSSION THREAD

**Discussion by zachweix** [Laravel-Backpack/community-forum#1145] [Q&A (Help)]
There are times when I want there to be no limit for a string in columns in the show operation. I accomplish this by just setting the limit to a very high number (e.g. 10000). While I know it's a minor operation for the `Str::limit($column['value'], $column['limit'], '…')` function to run, I think it would be more efficient to not run it at all. I am happy to go through all the columns and basically check if `$column['limit'] === false` or perhaps `$column['limit'] === null` then to not run it at all, but I want to know if that's something Backpack is interested in before taking care of that

**Comment by pxpm** [comment]
Hey @zachweix thanks for the suggestion. 

Indeed this `limit` is something we should probably have a look at. For me, there are two issues we can tackle if we mess with limits: 
1 - what you mentioned, `limit => false` no limit is applied, otherwise keep the default behavior. 
2 - `limitTerminator` - at the moment we have `...` (elipsis) hardcoded. I think there is an opportunity here to make this bit more configurable, like accepting a closure or something, so you can do something like:
```php
limitTerminator => fn($entry) => '... <a href="'.backpack_url('something/'.$entry->id.'/edit).'">(read more)</a>'
```
What do you think ?

**Comment by zachweix** [comment]
I think it would have to be either a closure or text. But if anyone wants to have a different default, I can also add a config to have a different default terminator. If you're ok with this, I can do this for all non-pro columns (I don't have access to the pro codebase 🙂)

**Comment by zachweix** [comment]
Any thoughts on this, is this something I can try to work on now?

**Comment by pxpm** [comment]
Hey @zachweix sure, I've talked this with the team, the only feedback I got was to switch `limitTerminator` to `limitSuffix`

![image](https://github.com/user-attachments/assets/0d9c276b-e138-40c5-8446-807f388bedff)

Do it for one or two columns, we iterate the DX, check if it does what we want and how we want, and then we propagate the change to all other columns. 

Sounds good ? Thanks for your time and for taking care of this 🙏
